index.d.ts

```diff
@@ -0,0 +1,60 @@
+export interface Action {
+  type: string;
+}
+
+export type Reducer<S> = (state: S, action: Action) => S;
+
+export type Dispatch = (action: any) => any;
+
+export interface MiddlewareAPI<S> {
+  dispatch: Dispatch;
+  getState: () => S;
+}
+
+export interface Middleware {
+  <S>(api: MiddlewareAPI<S>): (next: Dispatch) => Dispatch;
+}
```

Comment:
The problem is — `thunkMiddleware` is standalone and it can't have knowledge of what `MyStore` is.